crates/core/tedge/src/bridge/common_mosquitto_config.rs

```diff
@@ -110,6 +110,10 @@ impl Default for CommonMosquittoConfig {
 
 impl CommonMosquittoConfig {
     pub fn serialize<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
+        // Enable persistence as thin-edge.io heavily relies on retained messages and
+        // if mosquitto persistence is not configured, then it can lead to unexpected behavior
+        writeln!(writer, "persistence true")?;
```

Comment:
Beware that the file `/etc/tedge/mosquitto-conf/tedge-mosquitto.conf` is populated only if there is at least one cloud connected.

I would consider to improve the `tedge refresh-bridges` command to populate this file even if there is no bridge configured. I.e. moving [these 2 lines](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge/src/cli/refresh_bridges.rs#L38) before [that one](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/tedge/src/cli/refresh_bridges.rs#L33)